### ISSUE TITLE
Problem with Spark slaves connection

### ISSUE THREAD

**Issue by hyviquel** [BD2KGenomics/cgcloud#235]
Spark slaves do not connect alone to the master after creation of the cluster.
I need to restart spark using the following command to make them connect

```
/opt/sparkbox/spark/sbin/stop-all.sh
/opt/sparkbox/spark/sbin/start-all.sh
```

I don't know if this is the problem but I update spark to version 2.0.1 and hadoop to version 2.7.3

**Comment by hannes-ucsc** [comment]
I'm inclined to think that this has to do with your changes to the code. The tests would break if this was a problem on a vanilla master.

**Comment by hannes-ucsc** [comment]
If you want help with this, I propose that you file a PR with the changes you mentioned (upgrading Spark and Hadoop). We can then see if this replicates on our continuous integration server.

**Comment by hyviquel** [comment]
fixed by 1641103cdb7348a2c1183a16ad29091c947a2c47
